src/main/java/org/spongepowered/api/resource/metadata/MetadataSection.java

```diff
@@ -22,38 +22,43 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.asset;
+package org.spongepowered.api.resource.metadata;
 
-import org.spongepowered.plugin.PluginContainer;
-
-import java.util.Optional;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.persistence.DataQuery;
+import org.spongepowered.api.data.persistence.DataView;
 
 /**
- * The AssetManager offers a convenient way to easily retrieve resources from
- * Sponge {@link PluginContainer plugins}. The asset manager will attempt to find the
- * asset of the specified name at: <code>assets/&lt;plugin_id&gt;</code>
+ * A meta section is a well-known resource metadata object. Metadata is typically
+ * stored in a file with the resource and {@code .mcmeta} appended to the name.
+ * The file contains a json object with each key associated with a section.
+ *
+ * @param <T> The object type
  */
-public interface AssetManager {
+public interface MetadataSection<T> {
```

Comment:
The thing that strikes me about this is that it's basically wrapping data stuff to access the metadata - we have a query and a deserialise method. Can this be `DataSerializable`, or perhaps more to the point, should it?

I'm not arguing for this case so much, more that I'm seeing parallels that we may want to exploit.